crowbar_framework/app/models/api/upgrade.rb

```diff
@@ -151,8 +151,12 @@ def adminrepocheck
               missing_repo_arch = v[:repos].keys.first.to_sym
               v[:repos][missing_repo_arch][:missing]
             end.flatten.join(", ")
-            upgrade_status.end_step(false, adminrepocheck: "Missing repositories: #{missing_repos}")
+            upgrade_status.end_step(
+              false,
+              adminrepocheck: "Missing repositories: #{missing_repos}"
+            ) if upgrade_status.current_step == :admin_repo_checks
           else
+            next unless upgrade_status.current_step == :admin_repo_checks
```

Comment:
it returns with `ret` from the {}.tap block. for a similar example look at https://github.com/crowbar/crowbar-init/commit/c04698738fb754baa3cec375df94d96c0421ca94